### PR TITLE
Add Exports To Transparent Organisations & Refactor Exports Into a Dedicated Controller 

### PR DESCRIPTION
**Pull request by sampoder**
This PR removes the statements that hid the export feature when viewing transparent organisations. Most of it, however, is focused on adding a page to collect an email for large exports (those viewing transparent organisations aren't always signed-in).

I'd also appreciate if someone could do a quick security sanity check on this one.

Fixes #3371.

### CODE REVIEW

**Review by garyhtou — COMMENTED** [review]
Great start to this PR. Right now, we're handling exports in the `TransactionsController` and now we're adding to the `StaticPagesController` (which I totally understand why). However, I'm nervous about spreading the export logic throughout the codebase more. I would prefer if we focused on refactoring to an `Export` model and controller (and maybe a `FinanicalExport` model) before we add this feature

**Comment by sampoder** [comment]
Alrighty @garyhtou! This PR has ballooned a little but.... this should now refactor everything into an export controller and into export jobs so there shouldn't be any tie back into the transactions controller. 

One thing is that I didn't end up needing a model for what we currently have.

File: app/controllers/static_pages_controller.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 class StaticPagesController < ApplicationController
   skip_after_action :verify_authorized # do not force pundit
-  skip_before_action :signed_in_user, only: [:branding, :brand_guidelines, :faq]
+  skip_before_action :signed_in_user, only: [:branding, :brand_guidelines, :faq, :email_transparent_export]
```
Comment (garyhtou) [inline app/controllers/static_pages_controller.rb]:
```suggestion
  skip_before_action :signed_in_user, only: [:branding, :brand_guidelines, :faq]
```

File: app/controllers/transactions_controller.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 class TransactionsController < ApplicationController
   skip_before_action :signed_in_user
+  skip_after_action :verify_authorized, only: [:export]
```
Comment (garyhtou) [inline app/controllers/transactions_controller.rb]:
```suggestion
```

File: app/mailers/canonical_transaction_mailer.rb [hunk 3]
```diff
@@ -1,17 +1,18 @@
 # frozen_string_literal: true
 
 class CanonicalTransactionMailer < ApplicationMailer
```
Comment (garyhtou) [inline app/mailers/canonical_transaction_mailer.rb]: Can we move this to `ExportMailer`?

File: app/views/exports/collect_email.erb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+<div class="container container--sm card py3 md-p3 lg-p4 mt2 mt3 md-mt4 overflow-visible">
+  <h3 class="mt2 md-mt0">
+    This export is too big, so we'll send you an email when it's ready.
+  </h3>
+  <p>
+    We don't store emails submitted here in any of our databases, once you've recieved your file,
+    you won't hear from us again. 
+  </p>
+  <%= form_with(model: nil, local: true, method: "get", url: "/exports/#{@event_slug}.#{@file_extension}") do |form| %>
```
Comment (garyhtou) [inline app/views/exports/collect_email.erb]: See if we can use a URL helper here

Comment (garyhtou) [reply app/views/exports/collect_email.erb]: maybe something like this?
```ruby
export_exports_path(event: @event_slug, format: @file_extension)
```

File: app/controllers/exports_controller.rb [hunk 5]
```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+require "csv"
+
+class ExportsController < ApplicationController
+  skip_before_action :signed_in_user
+  skip_after_action :verify_authorized, only: [:export, :collect_email]
+
+  def export
```
Comment (garyhtou) [inline app/controllers/exports_controller.rb]: rename `export` action to `transactions`

**Review by garyhtou — COMMENTED** [review]
great job refactoring! just a couple things left

File: app/views/exports/collect_email.erb (hunk 4 above)
Comment (sampoder) [reply app/views/exports/collect_email.erb]:
```suggestion
  <%= form_with(model: nil, local: true, method: "get", url: export_exports_path(event: @event_slug, format: @file_extension)) do |form| %>
```

File: app/controllers/exports_controller.rb [hunk 6]
```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+require "csv"
```
Comment (garyhtou) [inline app/controllers/exports_controller.rb]:
```suggestion
```

File: app/mailers/export_mailer.rb [hunk 7]
```diff
@@ -1,17 +1,18 @@
 # frozen_string_literal: true
 
-class CanonicalTransactionMailer < ApplicationMailer
-  def export_ready(event:, user:, mime_type:, title:, content:)
+class ExportMailer < ApplicationMailer
+  def export_ready(event:, email:, mime_type:, title:, content:)
     @event = event
     @user = user
```
Comment (garyhtou) [inline app/mailers/export_mailer.rb]:
```suggestion
```

**Review by garyhtou — APPROVED** [review]

**Pull request merged**